msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833526.java

```diff
@@ -0,0 +1,175 @@
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.UiResponse;
+import com.microsoft.identity.internal.testutils.labutils.LabConfig;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+
+// Broker Auth for MDM account
+public class TestCase833526 extends AbstractAcquireTokenNetworkTest {
+
+    @Test
+    public void test_833526() throws InterruptedException {
+        final String username = mLoginHint;
+        final String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        final AcquireTokenParameters parameters = new AcquireTokenParameters.Builder()
+                .startAuthorizationFromActivity(mActivity)
+                .withLoginHint(username)
+                .withScopes(Arrays.asList(mScopes))
+                .withCallback(cancelInteractiveCallback(latch))
+                .withPrompt(Prompt.SELECT_ACCOUNT)
+                .build();
+
+        // start interactive token request in MSAL
+        // we expect to see enroll page as device is not enrolled
+        final InteractiveRequest interactiveRequest = new InteractiveRequest(
+                mApplication,
+                parameters,
+                new OnInteractionRequired() {
+                    @Override
+                    public void handleUserInteraction() {
+                        final PromptHandlerParameters promptHandlerParameters = PromptHandlerParameters.builder()
+                                .prompt(PromptParameter.SELECT_ACCOUNT)
+                                .loginHintProvided(true)
+                                .sessionExpected(false)
+                                .consentPageExpected(false)
+                                .speedBumpExpected(false)
+                                .broker(getBroker())
+                                .expectingNonZeroAccountsInBroker(false)
+                                .enrollPageExpected(true)
+                                // cancel enroll here to short circuit as enroll will be started manually from CP anyway
+                                .enrollPageResponse(UiResponse.DECLINE)
+                                .build();
+
+                        new AadPromptHandler(promptHandlerParameters)
+                                .handlePrompt(username, password);
+                    }
+                }
+        );
+
+        interactiveRequest.execute();
+        latch.await();
+
+        // enroll device with CP
+
+        final IMdmAgent mdmAgent = (IMdmAgent) mBroker;
+        mdmAgent.enrollDevice(username, password);
+
+        // SECOND REQUEST WITHOUT LOGIN HINT
```

Comment:
with login hint?

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1081/commits/0b9f6418d9893d1b5cffa16d24e14fd0dd989d94